orttraining/orttraining/test/python/orttraining_test_binary.py

```diff
@@ -0,0 +1,138 @@
+import pytest
+import torch
+import numpy as np
+import os
+import shutil
+
+from onnxruntime.training import _binary
+
+# Helper functions
+
+def equals(a, b):
+    if isinstance(a, dict):
+        for key in a:
+            if key not in b or not equals(a[key], b[key]):
+                return False
+        return True
+    else:
+        are_equal = a == b
+        return are_equal if isinstance(are_equal, bool) else are_equal.all()
+
+    return False
+
+def numpy_types(obj_value):
+    if not isinstance(obj_value, dict):
+        return type(obj_value).__module__ == np.__name__ or isinstance(obj_value, str)
+
+    for _, value in obj_value.items():
+        if not numpy_types(value):
+            return False
+    return True
+
+def get_dict(separated_key):
+    test_dict = {'int1':1, 'int2': 2, 'int_list': [1,2,3,5,6], 'dict1': {'np_array': np.arange(100), 'dict2': {'int3': 3, 'int4': 4}, 'str1': "onnxruntime"}, 'bool1': bool(True), 'int5': 5, 'float1': 2.345, 'np_array_float': np.array([1.234, 2.345, 3.456]), 'np_array_float_3_dim': np.array([[[1,2],[3,4]], [[5,6],[7,8]]])}
+    key = ''
+    expected_val = test_dict
+    for single_key in separated_key:
+        key += single_key + '/'
+        expected_val = expected_val[single_key]
+    return test_dict, {'key': key} if len(separated_key) > 0 else dict(), expected_val
+
+# Test fixtures
+
+@pytest.fixture(scope="function")
+def binary_test_setup():
+    binary_dir = os.path.abspath('binary_dir/')
+    if not os.path.exists(binary_dir):
+        os.makedirs(binary_dir, exist_ok = True)
+    pytest.binary_path = os.path.join(binary_dir, 'binary.ortcp')
+    yield 'binary_test_setup'
+    shutil.rmtree(binary_dir)
+
+@pytest.fixture(scope="function")
+def binary_parameterized_setup(request, binary_test_setup):
```

Comment:
how are this and binary_test_setup get invoked?

Reply:
the pytest fixture decorator invokes both. The unit test takes in ```binary_parameterized_setup``` as an argument, and since it is a pytest fixture, it gets invoked before running the test. Similarly, ```binary_parameterized_setup``` takes ```binary_test_setup``` as an argument and since it is also a fixture it gets invoked before ```binary_parameterized_setup``` is executed.